webapp/packages/plugin-datasource-transaction-manager/src/TransactionLogs/TransactionLogsDialog.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { observer } from 'mobx-react-lite';
+
+import {
+  Button,
+  CommonDialogBody,
+  CommonDialogFooter,
+  CommonDialogHeader,
+  CommonDialogWrapper,
+  Container,
+  Flex,
+  useAutoLoad,
+  useResource,
+  useTranslate,
+} from '@cloudbeaver/core-blocks';
+import { type ConnectionExecutionContext, ConnectionInfoResource, createConnectionParam } from '@cloudbeaver/core-connections';
+import type { DialogComponent } from '@cloudbeaver/core-dialogs';
+
+import { TransactionLogsTable } from './TransactionLogsTable/TransactionLogsTable.js';
+import { useTransactionLogs } from './useTransactionLogs.js';
+
+interface IPayload {
+  transaction: ConnectionExecutionContext;
+  onCommit: () => void;
+  onRollback: () => void;
+}
+
+export const TransactionLogsDialog: DialogComponent<IPayload> = observer(function TransactionLogsDialog(props) {
+  const translate = useTranslate();
+  const context = props.payload.transaction.context;
+  const connectionParam = context ? createConnectionParam(context.projectId, context.connectionId) : null;
+
+  const state = useTransactionLogs(props.payload);
+  const connectionInfoResource = useResource(useTransactionLogs, ConnectionInfoResource, connectionParam);
+  let title: string = translate('plugin_datasource_transaction_manager_logs');
+
+  if (connectionInfoResource.data?.name) {
+    title = `${title} (${connectionInfoResource.data.name})`;
+  }
+
+  useAutoLoad(TransactionLogsDialog, state);
+
+  function handleRollback() {
+    props.payload.onRollback();
+    props.resolveDialog();
+  }
+
+  function handleCommit() {
+    props.payload.onCommit();
+    props.resolveDialog();
+  }
```

Comment:
in case of error dialog will be closed, is it expected?

Reply:
We handle it on the onCommit level (behaviour the same as we press commit button in the top bar itself)